pkg/network/node/node.go

```diff
@@ -250,6 +252,73 @@ func GetLinkDetails(ip string) (netlink.Link, *net.IPNet, error) {
 	return nil, nil, ErrorNetworkInterfaceNotFound
 }
 
+func (node *OsdnNode) validateMTU() error {
+	klog.V(2).Infof("Checking default interface MTU")
+
+	// Get the interface with the default route
+	// TODO(cdc) handle v6-only nodes
+	routes, err := netlink.RouteList(nil, netlink.FAMILY_V4)
+	if err != nil {
+		return fmt.Errorf("could not list routes while validating MTU: %v", err)
+	}
+	if len(routes) == 0 {
+		return fmt.Errorf("got no routes while validating MTU")
+	}
+
+	const maxMTU = 65536
+	mtu := maxMTU + 1
+	for _, route := range routes {
+		// Skip non-default routes
+		if route.Dst != nil {
+			continue
+		}
+		link, err := netlink.LinkByIndex(route.LinkIndex)
+		if err != nil {
+			return fmt.Errorf("could not retrieve link id %d while validating MTU", route.LinkIndex)
+		}
+
+		newmtu := link.Attrs().MTU
+		if newmtu > 0 && newmtu < mtu {
+			mtu = newmtu
+		}
+	}
+	if mtu > maxMTU {
+		return fmt.Errorf("unable to determine MTU while performing validation")
+	}
+
+	if int(node.mtu) > mtu+50 {
+		resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+			klog.V(2).Infof("Configured node MTU is more than default interface MTU plus VXLAN overhead, tainting node...")
+			kNode, err := node.kClient.CoreV1().Nodes().Get(node.hostName, metav1.GetOptions{})
+			const MTUTaintKey string = "network.openshift.io/mtu-too-small"
+
+			if err != nil {
+				return fmt.Errorf("could not get Kubernetes Node object by hostname: %v", err)
+			}
+
+			_, _, err = taints.AddOrUpdateTaint(kNode, &corev1.Taint{Key: MTUTaintKey, Value: "value", Effect: "NoSchedule"})
+
+			if err != nil {
+				return fmt.Errorf("could not taint the node with key %s: %v", MTUTaintKey, err)
+			}
+
+			kNode, err = node.kClient.CoreV1().Nodes().Update(kNode)
```

Comment:
Probably want to do a patch here, see https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/writer.go#L193 for a sample implementation.